tests/marqo_test.py

```diff
@@ -185,87 +195,99 @@ def warm_request(self, func, *args, **kwargs):
         if settings_dict and kwargs:
-            raise ValueError("Cannot provide both settings_dict and kwargs.")
-        elif settings_dict or kwargs:
-            index_name = f"{index_name}-{create_settings_hash(settings_dict, **kwargs)}"
-        if settings_dict:
-            settings_dict.update({
-                "inference_type": "marqo.CPU.large", "storage_class": "marqo.basic"
-            })
-
-        if len(index_name) > 32:
-            raise ValueError(f"Index name {index_name} is too long.")
-        try:
-            status = client.http.get(f"indexes/{index_name}/status")["index_status"]
-            if status == "CREATING":
-                cloud_wait_for_index_status(client.http, index_name, "READY")
-            elif status != "READY":
-                raise RuntimeError(f"Indexes must be precreated during cloud tests. Index {index_name} is not ready. "
-                                   f"Please specify it's settings in scripts/populate_all_indices.py")
-        except (MarqoWebError, TypeError) as e:
-            raise RuntimeError(f"Indexes must be precreated during cloud tests. Index {index_name} is not ready. "
-                               f"Please specify its settings in scripts/populate_all_indices.py")
-        self.cleanup_documents_from_index(index_name)
+            raise ValueError("Only one of settings_dict and kwargs can be specified")
+        kwargs = kwargs or {}
+        client.create_index(index_name, settings_dict=settings_dict, **kwargs)
         return index_name
 
-    def create_test_index(self, index_name: str, settings_dict: dict = None, **kwargs):
-        """Create a test index with the given name and settings and triggers specific logic if index is cloud index"""
-        client = marqo.Client(**self.client_settings)
-        if client.config.is_marqo_cloud:
-            index_name = self.create_cloud_index(index_name, settings_dict, **kwargs)
-        else:
-            client.create_index(index_name, settings_dict=settings_dict, **kwargs)
-        return index_name
+    def mark_for_cleanup_and_add_documents(self, index_name: str, documents: list, *args,   **kwargs):
+        """Add documents to index and mark for cleanup after test is run."""
+        if kwargs.get("get_ix_name"):
+            return index_name
+        unclosed_mocks = {}
+        # print out all active mocks
+        for active_mock in mock._patch._active_patches:
+            unclosed_mocks[active_mock.kwargs['side_effect']] = active_mock.kwargs['side_effect']([], get_ix_name=True)
+        print(f"unclosed mocks: {unclosed_mocks}")
+        self.add_documents_and_mark_for_cleanup_patch.stop()
+        res = self.client.index(index_name).add_documents(documents, *args, **kwargs)
+        self.add_documents_and_mark_for_cleanup_patch.start()
+        res_list = [res] if type(res) is not list else res
+        for result_to_process in res_list:
+            if not result_to_process['errors']:
```

Comment:
you need to handle the case when there is one document that couldn't be indexed because of an error but some other docs _were_ successfully indexed

Reply:
at least in the current test suite there is not a single occurence of such case, and I couldn't reproduce I don't think we need to account for such case as we still have the default cleanup that uses search which would occur if numberOfDocs is not 0 after this one.